plinth/modules/dynamicdns/templates/dynamicdns_configure.html

```diff
@@ -0,0 +1,168 @@
+{% extends "base.html" %}
+{% comment %}
+#
+# This file is part of Plinth.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see http://www.gnu.org/licenses/.
+#
+{% endcomment %}
+
+{% load bootstrap %}
+
+{% block content %}
+
+  <form class="form" method="post">
+    {% csrf_token %}
+    <div id='dynamicdns-no-js'>
+      You have disabled Javascript. Dynamic form mode is disabled and 
+      some helper functions may not work <br>
+      (but the main functionality should work)
+      <br><hr>
+    </div>
+
+    {{ form|bootstrap }}
+
+    <input type="submit" class="btn btn-primary" value="Update setup"/>
+
+  </form>
+
+{% endblock %}
+
+{% block page_js %}
+  <script type="text/javascript">
+
+    (function($) {
+      var SELFHOST = 'https://carol.selfhost.de/update?username=<User>&' +
+                     'password=<Pass>&myip=<Ip>'
+      var NOIP     = 'http://dynupdate.no-ip.com/nic/update?hostname=' +
+                     '<Domain>&myip=<Ip>'
+      var FREEDNS  = 'https://freedns.afraid.org/dynamic/update.php?' +
+                     '_YOURAPIKEYHERE_'
+
+      //hide javascript warning
+      $('#dynamicdns-no-js').hide();
+      //hide ALL form fields
+      $('.form-group').hide();
+      //show the enable checkbox
+      $('#id_dynamicdns-enabled').closest('.form-group').show();
+      if ($('#id_dynamicdns-enabled').prop('checked')) {
+      //show all form fields
+        show_all();
+        //set the selectbox to the last configured value
+        select_service();
+      }
+
+      $('#id_dynamicdns-enabled').change(function() {
+        if ($('#id_dynamicdns-enabled').prop('checked')) {
+          show_all();
+          if ($("#id_dynamicdns-service_type option:selected").text() == "GnuDIP") {
+              set_gnudip_mode()
+            }else{
+              set_update_url_mode();
+            }
+        } else {
+          $('.form-group').hide();
+          $('#id_dynamicdns-enabled').closest('.form-group').show();
+        }
+      });
+
+      $('#id_dynamicdns-service_type').change(function() {
```

Comment:
It would be better (faster and better readable) to get the service type value only once and reuse it then, like
`var service = $("#id_dynamicdns-service_type option:selected").text()`
and then `if (service == "noip.com") {`